Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyAssertionOptions.cs

```diff
@@ -435,6 +435,7 @@ public TSelf Using(IEquivalencyStep equivalencyStep)
         /// </summary>
         public TSelf WithStrictOrdering()
         {
+            orderingRules.Clear();
```

Comment:
❓ Was `orderingRules` ever non-empty before this PR? Does this affect existing tests in any way?

Reply:
@jnyrup 
In case of `SelfReferenceEquivalencyAssertionOptions`, `OrderingRules` always contained the Byte-Ordering rule (see [here](https://github.com/fluentassertions/fluentassertions/blob/7738a652886af2e2ca0918ce71b197f7e06d8441/Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyAssertionOptions.cs#L64)).

This change did not make any existing tests fail.

As I've now just found out is that determining whether ordering should be `Strict` or `NotStrict` works differently than I thought. I thought the last rule takes priority, but that's not the case. Ordering remains `NotStrict` if the `OrderingRuleCollection` a rule which determines the member to be `NotStrict`, even if there's a newer rule that says "everything should be strict".
Hence calling `OrderingRuleCollection.Clear()` does affect the behavior. `WithStrictOrdering` calling `Clear()` overrides pre-existing `WithoutStrictOrderingFor` rules. Otherwise (not calling `Clear()`) `WithoutStrictOrderingFor` will remain effective.

Hence I've smuggled in a breaking change through the backdoor...

For reference, my design choice was based on the following understand (which I have found to be wrong):

It's my believe (i.E. i don't *know" for sure) that this should work. There's only two types of rules which can be contained in the list: those which require strict ordering and does which require non-strict ordering. Furthermore the last added rule wins. So if there's multiple contradictory rules for a member the last added one wins. Since `WithStrictOrdering()` and `WithoutStrictOrdering()` affect *all* members and - at this point - is the latest addition it will always win. All previously added rules become superfluous at this point.

Of course it would also have been *correct* (but not that nice to debug and not so performant) to just skip `orderingRules.Clear()`. `WithoutStrictOrdering()` would have, however, required a `MatchAllOrderingRule` counterpart which returns `OrderStrictness.NonStrict`.